lib/webrick/httpresponse.rb

```diff
@@ -268,11 +287,10 @@ def setup_header() # :nodoc:
       elsif %r{^multipart/byteranges} =~ @header['content-type']
         @header.delete('content-length')
       elsif @header['content-length'].nil?
-        if @body.respond_to? :readpartial
-        elsif @body.respond_to? :call
-          make_body_tempfile
```

Comment:
Not using a tempfile seems to break backwards compatibility.  Consider a callable body that opens a database connection or transaction.  Not buffering would open things up to denial of service via slow client attacks.  Can non-buffering mode be made opt-in?

Reply:
This was actually a feature added in <https://github.com/ruby/webrick/commit/4efbbf3be7f1ff783d8c88bd7b8fb1f61a9e29c5> to improve compatibility between HTTP/1.0 and HTTP/1.1 proxy but this PR broke streaming support (which based on the PR appears to have been working previously).

I think the expectation is that if you provide a body proc, it should receive the socket of the underlying connection.

Regarding a denial of service attack, I don't think anyone should be using WEBrick in a way which this is an issue. I think the PR which changed this behaviour was faulty.

Anyone who wants to have a buffered response should simply assign a string or StringIO to the response body.

Reply:
Considering the unbuffered->buffered changed in a minor release (1.4->1.5), changing it back in a minor release sounds fine.  We need to make sure be bump the minor before the next release.  I would like to document the change in a Changelog, but it looks like WEBrick does not have a Changelog.  Might be worth adding a Changelog, at least for changes since 1.7.0, but that is out of scope for this PR.

Reply:
I believe we are documenting changes using GitHub releases. I think that's a pretty good system.